ArduPlane/mode_qstabilize.cpp

```diff
@@ -11,29 +11,32 @@ bool ModeQStabilize::_enter()
 
 void ModeQStabilize::update()
 {
-    // set nav_roll and nav_pitch using sticks
+    // fetch normalized roll and pitch inputs
     // Beware that QuadPlane::tailsitter_check_input (called from Plane::read_radio)
     // may alter the control_in values for roll and yaw, but not the corresponding
     // radio_in values. This means that the results for norm_input would not necessarily
     // be correct for tailsitters, so get_control_in() must be used instead.
-    // normalize control_input to [-1,1]
-    const float roll_input = (float)plane.channel_roll->get_control_in() / plane.channel_roll->get_range();
-    const float pitch_input = (float)plane.channel_pitch->get_control_in() / plane.channel_pitch->get_range();
+    const float roll_in_norm = plane.channel_roll->get_control_in() / plane.channel_roll->get_range();
```

Comment:
You need to keep the float cast here, otherwise you are doing all the math in int16_t, and will find the results to be really bad.

Reply:
oops